experimental/QuadFormAndIsom/src/enumeration.jl

```diff
@@ -0,0 +1,952 @@
+  # Otherwise, their localisations at p must be rationally equal.
+  if g == 0
+    return local_symbol(AperpB, p) == local_symbol(C, p)
+  elseif excess(local_symbol(AperpB, p)) != excess(local_symbol(C, p))
+    return false
+  end
+
+  if !(Hecke.divides(scale(AperpB), scale(C))[1] && Hecke.divides(p*level(C), level(AperpB))[1])
+    return false
+  end
+
+  # At this point, if C is unimodular at p, the glueing condition is equivalent to have 
+  # an anti-isometry between the p-part of the (quadratic) discriminant forms of A and B
+  qA = discriminant_group(A)
+  qB = discriminant_group(B)
+  if !Hecke.divides(numerator(det(C)), p)[1]
+    return is_anti_isometric_with_anti_isometry(primary_part(qA, p)[1], primary_part(qB, p)[1])[1]
+  end
+
+  l = valuation(level(C), p)
+  Ap = local_symbol(A, p)
+  Bp = local_symbol(B, p)
+  a_max = symbol(Ap, l+1)[2]
+  b_max = symbol(Bp, l+1)[2]
+  # For the glueing, rho_{l+1}(A_p) and rho_{l+1}(B_p) are anti-isometric, so they must have the
+  # same order
+  if a_max != b_max
+    return false
+  end
+  
+  # Since p^l*A^\vee/A is in the glue, its order is less than the order of the glue
+  if g < a_max
+    return false
+  end
+
+  a1 = symbol(Ap, 1)[2]
+  a2 = rank(Ap) - a1 - symbol(Ap, 0)[2]
+  b1 = symbol(Bp, 1)[2]
+  b2 = rank(Bp) - b1 - symbol(Bp, 0)[2]
+
+  ABp = symbol(local_symbol(AperpB, p))
+  Cp = local_symbol(C, p)
+
+  if a_max == g
+    if length(symbol(Ap)) > 1
+      Ar = ZZLocalGenus(p, symbol(Ap)[1:end-1])
+    else
+      Ar = genus(matrix(ZZ,0,0,[]), p)
+    end
+   
+    if length(symbol(Bp)) > 1
+      Br = ZZLocalGenus(p, symbol(Bp)[1:end-1])
+    else
+      Br = genus(matrix(ZZ, 0, 0, []), p)
+    end
+  
+    ABr = direct_sum(Ar, Br)
+
+    for i = 0:l-1
+      s1 = symbol(ABr, i)
+      s2 = symbol(Cp, i)
+      if s1[2] != s2[2]
+        return false
+      end
+      if p == 2 && s1[4] > s2[4]
+        return false
+      elseif p != 2 && s1[3] != s2[3]
+        return false
+      end
+    end
+  end
+
+  s3 = symbol(local_symbol(C, 2))[end]
+  if p == 2
+    s1 = symbol(Ap, s3[1]+1)
+    s2 = symbol(Bp, s3[1]+1)
+    if s1[4] != s2[4]
+      return false
+    end
+  end
+  Cp = symbol(Cp)
+  for s in Cp
+    s[1] += 2
+  end
+  Cp = ZZLocalGenus(p, Cp)
+  
+  if !represents(local_symbol(AperpB, p), Cp)
+    return false
+  end
+  if !represents(C, AperpB)
+    return false
+  end 
+
+  qA, qB, qC = discriminant_group.([A, B, C])
+  spec = (p == 2) && (_is_free(qA, p, l+1)) && (_is_free(qB, p, l+1)) && (_is_even(qC, p, l))
+  rA = Oscar._rho_functor(qA, p, l+1)
+  rB = Oscar._rho_functor(qB, p, l+1)
+  if spec
+    return is_anti_isometric_with_anti_isometry(rA, rB)[1]
+  else
+    return Oscar._anti_isometry_bilinear(rA, rB)[1]
+  end
+end
+
+function is_admissible_triple(A::T, B::T, C::T, p::Integer) where T <: Union{ZZLat, ZZLatWithIsom}
+  L = ZZGenus[genus(D) for D = (A, B, C)]
+  return is_admissible_triple(L[1], L[2], L[3], p)
```

Comment:
```suggestion
  return is_admissible_triple(genus(A), genus(B), genus(C), p)
```
This allocates unnecessary memory.